File: C++ Subsystems/04-operator-overloading-and-smart-pointers/Student_Database/Student.cpp
#include "Student.h"

void Student::input() {
    cout << "Enter Roll Number: ";
    cin >> rollNo;

    cout << "Enter Name: ";
    cin >> name;

    cout << "Enter marks of 5 subjects:\n";
    for(int i=0;i<5;i++) cin >> marks[i];
}

void Student::display() {
    cout << "Roll No: " << rollNo << endl;
    cout << "Name: " << name << endl;

    cout << "Marks: ";
    for(int i=0;i<5;i++) cout << marks[i] << " ";
    cout << endl;
}

File: Java Subsystems/01-abstract-banking-and-exception-pipelines/q4/Student.java
public class Student {
    private int rollNo;
    private String name;
    private int[] marks; // 5 subjects

    public Student(int rollNo, String name, int[] marks) {
        this.rollNo = rollNo;
        this.name = name;
        this.marks = marks;
    }

    public int getRollNo() { return rollNo; }
    public String getName() { return name; }
    public int[] getMarks() { return marks; }

    public double getAverage() {
        int sum = 0;
        for (int m : marks) sum += m;
        return sum / (double) marks.length;
    }

    public void display() {
        System.out.print("Roll No: " + rollNo + " | Name: " + name + " | Marks: ");
        for (int i = 0; i < marks.length; i++) {
            System.out.print("S" + (i + 1) + "=" + marks[i]);
            if (i < marks.length - 1) System.out.print(", ");
        }
        System.out.printf(" | Average: %.2f%n", getAverage());
    }
}
